### ISSUE TITLE
Missing information in readme on how to use the functionMocks section

### ISSUE THREAD

**Issue by peterhav** [willhlaw/firestore-security-tests#2]
Hello,

I'm trying to use this node to perform some test on my Firebase rules. Simple tests seem to work fine, but when creating some more advanced tests, I run into the problem that the documentation is to limited at this moment.
I need more information on how to use the functionMocks section inside the testcases.
Information that is required:

1. What should be the content of the args: '[{ exact_value:' part
2. What should be the content of the result: '{ value: { data: {' part

I have a test case that I expect to work, but can't get it to work in using the firestore-security-tests:

```
{
        expectation: 'ALLOW', //READ OWNED EVENT DATA
        request: {
          auth: {
            uid: '7QLCpgSZ5CdaVhj52GC50jhe1o02'
          },
          path: '/databases/(default)/documents/event/qOdmBUv4AeOIHSd5K1PH',
          method: 'get'
        },
        functionMocks: [
          {
            function: 'get',
            args: [{ exact_value: '/events/qOdmBUv4AeOIHSd5K1PH' }],
            result: { value: { data: { owner: '7QLCpgSZ5CdaVhj52GC50jhe1o02' } } }
          }
        ]
      }
```

The above should be working with the provided rules (it works runtime), can't get this to pass in the tests however. In my rules I defined that the events are readable when they have an attribute 'owner' that is equal to the uid.

**Comment by willhlaw** [comment]
@peterhav What do your rules look like under source.files.content?

What you have constructed looks correct and I was able to use it successfully in the index.spec.js file and run `yarn test` and it passes.

For [slightly] more info on `functionMocks`, see https://www.any-api.com/googleapis_com/firebaserules/docs/Definitions/FunctionMock.

**Comment by peterhav** [comment]
The rules used are the following:

```
service cloud.firestore {
  match /databases/{database}/documents {
  
  	allow read, write: if request.auth != null; //only authorized users can read and write things
    
    // Collection named "users", document named after the userId
    match /users/{userId} { 
      allow read, write: if isCurrentUser( userId );
  	}
		
		match /events/{eventId} {
			allow create: if isCurrentUser( request.resource.data.owner );
      allow update: if isCurrentUser( resource.data.owner ) && isCurrentUser( request.resource.data.owner );
      allow delete, read: if isCurrentUser( resource.data.owner );
     
      match /{allChildren=**} {
           // Only signed in users can read this data
           allow read, write: if request.auth.uid != null;

      }
		}
  }
  
  // Check if a particular user made a request
	function isCurrentUser( userId ) {
  	return request.auth.uid == userId;
	}
}
```

**Comment by willhlaw** [comment]
@peterhav, I just looked into this and have concluded that `resource` is not available currently in Google's `firebaserules.projects.test` API. It would make sense because the API runs the rules against simulated logic. Thus, they do not have any idea of what the existing `resource` would look like. As an aside in case you were wondering, the creds you provide are only to be able to communicate to Google's API; not (yet) to your actual Firestore database. Google has updated their [draft document](https://docs.google.com/document/d/1rc_S5RrnxbEUezVN6uwLFNdttdeZ1o9ugQC_UarXRFI/edit#) to describe how to use the `firebaserules.projects.releases.list` API to run the testCases against the live rules, but I haven't integrated that into this project yet. So I wonder if `resource` would be available then. Once I or somebody adds the releases.list functionality, I'll update this issue on whether or not they implement the `resource` object.

This takes us back to your original question about how to use the functionMocks part. The functions to mock are only the "Service Defined" functions `exists()` and `get()` as defined in the Firstore docs (https://firebase.google.com/docs/firestore/reference/security/#service_defined).

Thus, you would have to replace `resource.data.owner` with `get(/databases/$(database)/documents/events/$(eventId)).data.owner` when testing the rules.

- See [working example](https://runkit.com/willhlaw/firestore-security-rules-get-resource-not-usable-example). 
- See [failure example](https://runkit.com/willhlaw/firestore-security-rules-get-resource-not-usable-fails-example) where using `resource` fails.

By the way @peterhav, I noticed that you are requesting `/event/` where your mock has it plural, `/events/`. In the above examples, I made everything consistent using the plural form, `/events/`.
